packages/libs/src/sdk/api/users/UsersApi.ts

```diff
@@ -188,4 +194,121 @@ export class UsersApi extends GeneratedUsersApi {
       ...advancedOptions
     })
   }
+
+  /**
+   * Downloads the sales the user has made as a CSV file
+   */
+  async downloadSalesAsCSVBlob(params: DownloadSalesAsCSVRequest): Promise<Blob> {
+    if (params.id === null || params.id === undefined) {
+      throw new runtime.RequiredError('id','Required parameter params.id was null or undefined when calling downloadSalesAsCSV.');
+    }
+
+    if (params.encodedDataMessage === null || params.encodedDataMessage === undefined) {
+        throw new runtime.RequiredError('encodedDataMessage','Required parameter params.encodedDataMessage was null or undefined when calling downloadSalesAsCSV.');
+    }
+
+    if (params.encodedDataSignature === null || params.encodedDataSignature === undefined) {
+        throw new runtime.RequiredError('encodedDataSignature','Required parameter params.encodedDataSignature was null or undefined when calling downloadSalesAsCSV.');
+    }
+
+    const queryParameters: any = {};
+
+    if (params.userId !== undefined) {
+        queryParameters['user_id'] = params.userId;
+    }
+
+    const headerParameters: runtime.HTTPHeaders = {};
+
+    if (params.encodedDataMessage !== undefined && params.encodedDataMessage !== null) {
+        headerParameters['Encoded-Data-Message'] = String(params.encodedDataMessage);
+    }
+
+    if (params.encodedDataSignature !== undefined && params.encodedDataSignature !== null) {
+        headerParameters['Encoded-Data-Signature'] = String(params.encodedDataSignature);
+    }
+
+    const host = await this.discoveryNodeSelectorService.getSelectedEndpoint()
+    const path = `/users/{id}/sales/download`.replace(`{${"id"}}`, encodeURIComponent(String(params.id)))
+    const url = `${host}${BASE_PATH}${path}`
+    const response = await fetch(url, { method: 'GET', headers: headerParameters })
+    return response.blob()
```

Comment:
I understand the confusion. I got to this point because it was a pain getting the generated api to return Blob, and when it did, the api endpoint itself didn't work.
So I created this function that 'overrides' it and returns a Blob to us.

Pretty frustrating that the API endpoint works, but the generated API does not allow us to get the data, hence the overriding. And I feel that I spent enough time on this for now. At least I want to merge a working change soon and do improvements as a follow-on.

Reply:
Makes sense. That does seem frustrating. Agreed we can get it merged for now and think of a better way to do it.
Maybe the route of generating a download link and then just navigating to it is less of a hassle. I don't know if fetching a blob into memory is worth the hassle :-)